### PR TITLE
Provider fuzzy query algorithm<FuzzyChineseEncryptAlgorithm> and test

### PR DESCRIPTION
**Pull request by gxxiong**
Ref #20435 .

Changes proposed in this pull request:
  -Provider fuzzy query algorithm<FuzzyChineseEncryptAlgorithm> and test
 

---

Before committing this PR, I'm sure that I have checked the following options:
- [X] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [X] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [X] I have passed maven check locally : `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [ ] I have made corresponding changes to the documentation.
- [X] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Fuzzy Chinese encrypt algorithm.
+ */
+@Getter
+public final class FuzzyChineseEncryptAlgorithm implements EncryptAlgorithm<Object, String> {
```
Comment (strongduanmu) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java:33]: Does this Algorithm support english?

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+                    + "药缨啸赚悔啄墅裤秋悸澄沿怂市猾螺英镊蟹示逗殊壶点厘侯这逆嚏烩肠豌虫蛋接脉拱芬与愈壁依殆肾铭摧火荣忠六逐莱罪匡熬藏眠义"
+                    + "宙无腰税诌檬航显涎跌赔涌阴敛务案摆吭阜篱簿懈蠢纲湃握翱延毕尤茄览阳桌植柯燕淮碟薛愤瓶痉档称绵照丝辖报什伯琉札册聘苟乔"
+                    + "蹄沧咀搀化拒雕哺眺匈腆廊诅音剁轰跟枷估实栗忍难揍山蔗砧含州准肇任艘砷葡汰屹铜吠津藐懒涩包廷轿播恤芋畅竣琼峰倔遂匙扦城"
+                    + "拯搐赂弃可贵缕霞周涡玉浑礁蜡巳太合哲堆弧恫郁壮观噎酉欢扒羞瞪僵颂皖跑朔仙屎脆硫膜砰磐租驴急做贸酌熏踞鬼虾途锥澈岔狙乳"
+                    + "单欲堡藤廉辱驹节隋馋淤灯踩泰成蝴顶吮母采柏咎责占页晨荡筛裳艇习远魂掳趾妒淋孺晴赵讶溜妇鸣拇骄于厢纺寞宋冗竖甫毛滩烘倚"
+                    + "廓截淆价很钠斋最毋锅标鲤捷初吞潘质寥覆疙膘宜克辫陛缩曰人府家瓢乃他美像井汛版甲镶宾催帐尸累遥唬奶亦轻七嫁匠侮洼扳链颠"
+                    + "靳脯昌官沾丙搂杜盛席消凛梯范颁籽病喊而辊抉坑燎粹账慎追均辜皆贝恿阻旨训柔赁堕陇匆浮量喧钳苹蕉进窘舜恳种伍险革胎瞎肚昨"
+                    + "棋晾扎煞铰诧汁破";
+    
+    private Map<Character, Integer> charIndexMap = new LinkedHashMap<>(4000);
```
Comment (strongduanmu) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java:118]: Please add final for charIndexMap.

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    
+    @Override
+    public void init(final Properties props) {
+        this.props = props;
+        if (props == null) {
```
Comment (strongduanmu) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java:123]: Please put null on the left hand.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java (hunk 1 above)
Comment (gxxiong) [reply features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/algorithm/FuzzyChineseEncryptAlgorithm.java:33]: Support all languages, Chinese parameters have been set, English and Latin do not need to be set, other languages do not set parameters will appear unusual words, does not affect the use

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**